react/src/tests/Emails-test.js

```diff
@@ -0,0 +1,669 @@
+    });
+
+    it("Sagas saveEmail", () => {
+
+       const generator = saveEmail(getState);
+       let next = generator.next(getState);
+       next.value.SELECT.selector = function (state) {
+            return state;
+       };
+       expect(next.value).toEqual(select(state => state));
+
+       const data = {
+                email: state.emails.email,
+                confirmed: false,
+                primary: false
+              };
+       const emails = generator.next(state, data);
+       expect(emails.value).toEqual(call(sendEmail, state.config, data));
+
+       next = generator.next(emails);
+       expect(next.value).toEqual(put(emails));
+    });
+
+    it("Sagas requestResendEmailCode", () => {
+
+       const generator = requestResendEmailCode(getState);
+       let next = generator.next();
+
+       next.value.SELECT.selector = function (state) {
+            return state;
+       };
+       expect(next.value).toEqual(select(state => state));
+
+        const data = {
+                email: state.emails.confirming
+              };
+
+        const resp = generator.next(state, data);
+        expect(resp.value).toEqual(call(requestResend, state.config, data));
+
+        next = generator.next(resp.value);
+        expect(next.value).toEqual(put(resp.value));
+    });
+
+});
+
+function setupComponent() {
+  const props = {
+    given_name: '',
+    surname: '',
+    display_name: '',
+    language: '',
+    handleSave: createSpy(),
+    handleChange: createSpy(),
+  };
+
+  const wrapper = shallow(<Emails {...props} />)
+
+  return {
+    props,
+    wrapper,
+  }
+}
+
+describe("Emails Component", () => {
+
+    it("Renders", () => {
+        const {wrapper, props} = setupComponent(),
+            form = wrapper.find('form'),
+            fieldset = wrapper.find('fieldset'),
+            email = wrapper.find('TextControl[name="email"]');
+        // TODO: not finished
+    });
+});
+
+import EmailsContainer from "containers/Emails";
+import { Provider } from 'react-redux';
+import { IntlProvider, addLocaleData } from 'react-intl';
+
+const messages = require('../../i18n/l10n/en');
+addLocaleData('react-intl/locale-data/en');
+
+const fakeStore = (state) => ({
+  default: () => {},
+  dispatch: createSpy(),
+  subscribe: createSpy(),
+  getState: () => ({ ...state })
+});
+
+
+describe("Emails Container", () => {
+  let fulltext,
+    email,
+    fulldom,
+    language,
+    mockProps,
+    wrapper,
+    dispatch;
+
+ beforeEach(() => {
+    const store = fakeStore({
+        emails: {
+            is_fetching: false,
+            failed: false,
+            error: '',
+            message: '',
+            resending: {
+              is_fetching: false,
+              failed: false,
+              error: {},
+              message: ''
+            },
+            confirming: '',
+            emails: [],
+            email: '',
+        },
+        is_fetching: true,
+      resending: {
```

Comment:
This seems wrong, there should be no `is_fetching` or `resending` keys at the root of the state object.
I guess `resending` should be `personal_data`, missing a `failed` key and with an extra `emails` key.